krystex/src/main/java/com/flipkart/krystal/krystex/node/Node.java

```diff
@@ -354,24 +354,41 @@ private void executeResolver(RequestId requestId, ResolverDefinition resolverDef
     if (resolverCommand instanceof SkipDependency) {
       if (dependencyValuesCollector.getOrDefault(requestId, ImmutableMap.of()).get(dependencyName)
           == null) {
+        /* This is for the case where for some resolvers the input has already been resolved but we
+        do need to skip them as well, as our current resolver is skipped.*/
+        Set<RequestId> requestIdSet = dependencyNodeExecutions.individualCallResponses().keySet();
+        for (RequestId depRequestId : requestIdSet) {
```

Comment:
Before this for loop we can add
```java
if(requestIdSet.isEmpty()){
   requestIdSet = Set.of(requestId.append("%s[%s]".formatted(dependencyName, 0)));
}
```

Reply:
I have initialised set with current resolver request id , and will add if we have other resolver outputs.